### DISCUSSION TITLE
None standard protocol over TLS connection

### DISCUSSION THREAD

**Discussion by johnmcvann** [mitmproxy/mitmproxy#4912] [Q&A] [answered]
Hello All 
I want to talk TLS to a mitmproxy and then to a server.
Once the TLS connection is established, we are trying to talk our own protocol over this connection.
I am getting an error with mitmproxy for this it is trying to parse the data as http, but it is not. 
Is this expected? 
Should the proxy not just pass through the data and not try and parse it? I have set the --rawtcp but it did not help.

mitmdump --rawtcp --set ssl_verify_upstream_trusted_ca=/tmp/my-ca.pem --set client_certs=/tmp/my-crt-key.key --ssl-insecure
Proxy server listening at http://*:8080
127.0.0.1:35802: clientconnect
 << Bad HTTP request line: b'CONNECT localhost:0 MYPROTOCOL/21.07'

the above setup works perfect when I do a HTTP request

**Comment by mhils** [comment, marked as answer]
mitmproxy generally tries to treat protocols that somewhat look like HTTP as HTTP. 
If this is not what you want, you can opt into generic TCP proxying by explicitly setting the `tcp_hosts` [option](https://docs.mitmproxy.org/dev/concepts-options/). :)
